source/common/upstream/upstream_impl.cc

```diff
@@ -1415,16 +1415,18 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
 
   // Remove hosts from current_priority_hosts that were matched to an existing host in the previous
   // loop.
-  for (auto itr = current_priority_hosts.begin(); itr != current_priority_hosts.end();) {
+  auto erase_from_itr = current_priority_hosts.end();
+  for (auto itr = current_priority_hosts.begin(); itr != erase_from_itr;) {
     auto existing_itr = existing_hosts_for_current_priority.find((*itr)->address()->asString());
 
     if (existing_itr != existing_hosts_for_current_priority.end()) {
       existing_hosts_for_current_priority.erase(existing_itr);
-      itr = current_priority_hosts.erase(itr);
+      *itr = *(--erase_from_itr);
     } else {
       itr++;
     }
   }
+  current_priority_hosts.erase(erase_from_itr, current_priority_hosts.end());
```

Comment:
Is this a related change? If it's a general iterator safety thing or the like, can you split to  distinct PR?

Reply:
"related" insofar as it "makes it faster"; the original thought was removing this o(n^2) operation would make a difference. Testing showed it didn't make a big one, but I still think its worth doing. "unrelated" to the "provide a control to ignore unused fields" goal, however. I'm happy to pull it out if it feels risky.

Reply:
Yeah, if you can split this into a later PR it would be great, thanks.